src/libraries/System.Net.Http/src/System/Net/Http/Headers/KnownHeaders.cs

```diff
@@ -107,6 +107,14 @@ internal static class KnownHeaders
         public static readonly KnownHeader XUACompatible = new KnownHeader("X-UA-Compatible");
         public static readonly KnownHeader XXssProtection = new KnownHeader("X-XSS-Protection", HttpHeaderType.Custom, null, new string[] { "0", "1", "1; mode=block" });
 
+        private static HttpHeaderParser? GetAltSvcHeaderParser() =>
+#if TARGET_BROWSER
```

Comment:
You could also use SocketsHttpHandler.IsSupported check

Reply:
That approach has 2 disadvantages:

1. We would need to suppress this:
```
C:\git\runtime\src\libraries\System.Net.Http\src\System\Net\Http\Headers\KnownHeaders.cs(112,13): 
    error CA1416: This call site is reachable on: 'Browser'. 'SocketsHttpHandler.IsSupported' is unsupported on: 'Browser'. [C:\git\runtime\src\libraries\System.Net.Http\src\System.Net.Http.csproj]
```
2. This way, on `Browser`, AltSvcHeaderParser isn't even compiled into the library. So if someone tries to take a reference to it in some other code, they will consciously need to add it back to be compiled during the `Browser` build.

Reply:
> We would need to suppress this:

You could use guarded supported attribute

> AltSvcHeaderParser isn't even compiled into the library.

It's the same it would be removed by linker during build and if someone adds it back it should trigger size regression warning

*Just showing an alternative way to implement this*